cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -57,28 +64,45 @@ public void run() {
     try {
       long tillTime = System.currentTimeMillis() - retentionDurationMs;
       final SetMultimap<String, Location> parentDirs = HashMultimap.create();
+      final Map<String, NamespaceId> namespacedLogBaseDirMap = new HashMap<>();
       fileMetaDataManager.cleanMetaData(tillTime,
                                         new FileMetaDataManager.DeleteCallback() {
                                           @Override
-                                          public void handle(Location location, String namespacedLogBaseDir) {
```

Comment:
@anwar6953 My understanding from what was conveyed earlier was that the operation such as reading/writing/deleting/exists etc checks needs to be wrapped. The comment says something else now. Can confirm what exactly needs to be done. It will be better if you add the requirement in the jira.